core/src/main/java/com/linecorp/armeria/common/stream/AbortingSubscriber.java

```diff
@@ -16,20 +16,19 @@
 
 package com.linecorp.armeria.common.stream;
 
+import java.util.function.Supplier;
+
 import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
 
 final class AbortingSubscriber<T> implements Subscriber<T> {
 
-    private static final AbortingSubscriber<Object> INSTANCE = new AbortingSubscriber<>();
+    private final Supplier<? extends Throwable> causeSupplier;
 
-    @SuppressWarnings("unchecked")
-    static <T> AbortingSubscriber<T> get() {
-        return (AbortingSubscriber<T>) INSTANCE;
+    AbortingSubscriber(Supplier<? extends Throwable> causeSupplier) {
+        this.causeSupplier = causeSupplier;
     }
```

Comment:
Could we continue using a singleton for `abort()`? We could create a new `AbostringSubscriber` only when a supplier is given.